### PR TITLE
feat: add support for ES2025 duplicate named capturing groups

### LINKED ISSUE #194: Support for duplicate named capturing groups
It has reached stage 4.
https://github.com/tc39/proposal-duplicate-named-capturing-groups

https://arai-a.github.io/ecma262-compare/?pr=2721

### PR DESCRIPTION
**Pull request by ota-meshi**
close #194

### CODE REVIEW

**Comment by RunDevelopment** [comment]
How does this handle backreferences? The [`resolved` field](https://github.com/eslint-community/regexpp/blob/a98196e3c17081753b1af4d01604a6c69a6366e6/src/ast.ts#L435) points to the referenced capturing group, but this is no longer unique.

**Comment by ota-meshi** [comment]
Thank you for letting me know! I think it's necessary as you say.
That should be treated as a breaking change, right?

**Comment by RunDevelopment** [comment]
> That should be treated as a breaking change, right?

Maybe. You currently implemented `resolved: CapturingGroup[]` which is a breaking change. However, we could also use the same trick you used for [Unicode set character classes in v-flag regexes](https://github.com/eslint-community/regexpp/blob/a98196e3c17081753b1af4d01604a6c69a6366e6/src/ast.ts#L225). So we could do something like this: 
```ts
type Backreference = AmbiguousBackreference | UnambiguousBackreference
interface BaseBackreference {
  type: "Backreference"
  // other props
  ambiguous: boolean
  resolved: CaptruingGroup | CaptruingGroup[]
}
interface AmbiguousBackreference extends BaseBackreference {
  ambiguous: true
  resolved: CaptruingGroup[]
}
interface UnambiguousBackreference extends BaseBackreference {
  ambiguous: false
  resolved: CaptruingGroup
}
```

The interface/property names could use some work, but I hope I communicated the idea. This wouldn't be a breaking change, because regexes without duplicate group names produce the same AST as before (except for the additional `ambiguous: false` property).

However, I'm not sure whether this AST is ideal. It's backwards compatible, but it's also a bit more tricky to understand.

**Comment by ota-meshi** [comment]
Thank you! That's a good idea!
I think it's a good thing that it avoids breaking changes since it is additional information that is not related to the AST.

File: test/fixtures/parser/literal/duplicate-named-capturing-group-invalid-2025.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "index": 27
+      }
+    },
+    "/(?<x>a)|(?<x>b)(?<x>c)/": {
+      "error": {
+        "message": "Invalid regular expression: /(?<x>a)|(?<x>b)(?<x>c)/: Duplicate capture group name",
+        "index": 21
+      }
+    },
+    "/(?:(?<x>a)|(?<x>b))(?<x>c)/": {
```
Comment (ota-meshi) [inline test/fixtures/parser/literal/duplicate-named-capturing-group-invalid-2025.json:19]: When I try it in Safari, it's not a syntax error. I may be misunderstanding something 🤔

Comment (ota-meshi) [reply test/fixtures/parser/literal/duplicate-named-capturing-group-invalid-2025.json:19]: If I try the swapped `/(?<x>c)(?:(?<x>a)|(?<x>b))/` in Safari, I will get a syntax error.
Is it a bug in Safari that `/(?:(?<x>a)|(?<x>b))(?<x>c)/` does not result in a syntax error? 🤔

Comment (ota-meshi) [reply test/fixtures/parser/literal/duplicate-named-capturing-group-invalid-2025.json:19]: It was a bug in Safari, which has now been fixed.

**Comment by ota-meshi** [comment]
@eslint-community/core-team Can someone please review this PR?

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Pull request merged**